src/customFileParser/unpackGeometry/MergedMesh.ts

```diff
@@ -0,0 +1,60 @@
+import { CompressedGeometryData, UncompressedValues } from '../sharedFileParserTypes';
+import PropertyLoader from '../PropertyLoader';
+import { MergedMeshGroup, MergedMesh } from '../../geometry/MergedMeshGroup';
+import SceneStats from '../../SceneStats';
+import Sector from './../../Sector';
+
+export default function unpackMergedMeshes(
+  rootSector: Sector,
+  uncompressedValues: UncompressedValues,
+  sectorPathToMergedMeshData: {[path: string]: CompressedGeometryData},
+  sceneStats: SceneStats,
+  treeIndexNodeIdMap: number[],
+  colorMap: THREE.Color[],
+) {
+  const data = new PropertyLoader(uncompressedValues);
+
+  for (const sector of rootSector!.traverseSectors()) {
+    sector.mergedMeshGroup = new MergedMeshGroup();
+    const geometryInfo = sectorPathToMergedMeshData[sector.path];
+    if (geometryInfo !== undefined) {
+      // count meshes per file Id
+      const meshCounts: any = {};
+      for (let i = 0; i < geometryInfo.count; i++) {
+        data.loadData(geometryInfo);
+        meshCounts[data.fileId] = meshCounts[data.fileId] ? meshCounts[data.fileId] : 0;
+        meshCounts[data.fileId]++;
+      }
+      geometryInfo.indexes.rewind();
+      geometryInfo.nodeIds.rewind();
+
+      // create merged meshes
+      const mergedMeshes: any = {};
+      Object.keys(meshCounts).forEach(fileId => {
+        if (meshCounts[fileId] !== 0) {
+          mergedMeshes[fileId] = new MergedMesh(meshCounts[fileId], parseInt(fileId, 10));
```

Comment:
`fileId` should be a number already?

Reply:
Javascript array keys get saved/iterated over as strings